### DISCUSSION TITLE
Android HttpClient

### DISCUSSION THREAD

**Discussion by sharpninja** [unoplatform/uno#8823] [Q&A]
Using latest dev build.  Every call to SendAsync, GetAsync, etc result in the same exception:

```cs
const string SERVICES_JSON_URL= "https://sharpninja.github.io/Activate/services.json";
HttpClient httpClient = new();

try
{
    json =
        await httpClient.GetStringAsync(SERVICES_JSON_URL);
}
catch (AggregateException ae) when (ae.InnerException is HttpRequestException)
{
}
catch (Exception ex)
{
}
finally
{
    if (httpClient is not null) httpClient.Dispose();
}
```

Am I missing something?

**Comment by jeromelaban** [comment]
Can you provide the exception, the targetframework you're using? Note that Uno does not influence the behavior of the underlying .NET framework.

**Reply by sharpninja** [reply]
it's the same project as the resources.  git should be up to date.  launch debug for android and find the code in the Load method of ServiceList.

**Reply by jeromelaban** [reply]
Sure, but give the all information in this thread anyways, not everyone understands you context. When you have an issue, make the context as explicit and concise as possible so we can help you.

**Reply by sharpninja** [reply]
1. Clone the repo using [this commit](https://github.com/sharpninja/Activate/tree/fb5fa71fea232328b0cf2957cec4b3efe69571b3).
2. Set startup project to Activate.Mobile.
3. Select and start and Android emulator, or start Windows Subsystem for Android.  Get the IP from the WSA app and then open the ADB command prompt in VS 2022.
```
adb connect <copied ip address>
```
3. Open ServiceList.cs under Activate.Shared in the Models folder.
4. Set breakpoints on lines 37, 45 and 50.
6. F5 to start debugging in the selected Android environment.  When the breakpoint hits line 37, use F10 to step until either line 45 or 50 is reached.  If line 50 is reached, you should see a NullReferenceException in the Locals window.

**Reply by jeromelaban** [reply]
Thank you. Please post the full exception that you're getting here first.

**Reply by sharpninja** [reply]
Full stack trace: Android

```
ex
{System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Net.Http.HttpClientHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpMessageInvoker.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<>n__0(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.GetStringAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   at Activate.Models.ServiceList.Load() in C:\GitHub\Activate\Activate\Activate.Shared\Models\ServiceList.cs:line 38}
    base: {System.SystemException}
```

**Reply by jeromelaban** [reply]
This looks like a framework issue. You'll probably need to create a very small repro which demos that particular issue, so it may be opened in the dotnet/runtime repository.
